### PR TITLE
feat(crc): Reject contract

### PR DESCRIPTION
**Pull request by Rikkiabb**
# \<Description\>
Rejection flow for parent B:
- Parent B has option to reject application or continue on landing page
- When Parent B decides to reject application the parent goes to a flow where the parent is asked if they are sure
- If they are sure an email is sent to parent A and the application goes into rejected state

Translations strings have been pushed to contentful.

## Screenshots / Gifs

Option for rejecting contract:
Screenshot 2021-05-05 at 14 54 28

Reject contract screen:
Screenshot 2021-05-05 at 16 06 10

Contract rejected screen:
Screenshot 2021-05-05 at 15 41 23

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { getSelectedChildrenFromExternalData } from '../..'
+
+const ContractRejected = ({ application }: CRCFieldBaseProps) => {
+  const { answers } = application
+  const selectedChildren = getSelectedChildrenFromExternalData(
+    application.externalData.nationalRegistry.data.children,
+    answers.selectedChildren,
+  )
+
+  const otherParentName = selectedChildren[0].otherParent.fullName
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx:14]: Not a part of this PR but I'm just wondering if we cant to create a utility for extracting `otherParent` from the application. Feel like we are doing this in a lot of places where we are using the `selectedChildren` only to get the `otherParent`.

File: libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { contractRejected } from '../../lib/messages'
+import { CRCFieldBaseProps } from '../../types'
+import { ContractRejectedContainer, DescriptionText } from '../components'
+import { getSelectedChildrenFromExternalData } from '../..'
+
+const ContractRejected = ({ application }: CRCFieldBaseProps) => {
+  const { answers } = application
+  const selectedChildren = getSelectedChildrenFromExternalData(
+    application.externalData.nationalRegistry.data.children,
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx:10]:
```suggestion
  const { answers, externalData } = application
  const selectedChildren = getSelectedChildrenFromExternalData(
    externalData.nationalRegistry.data.children,
```

File: libs/application/templates/children-residence-change/src/fields/ParentBContractRejected/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import { contractRejected } from '../../lib/messages'
+import { CRCFieldBaseProps } from '../../types'
+import { ContractRejectedContainer, DescriptionText } from '../components'
+
+const ParentBContractRejected = ({}: CRCFieldBaseProps) => {
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/ParentBContractRejected/index.tsx:6]:
```suggestion
import { ContractRejectedContainer, DescriptionText } from '../components'

const ParentBContractRejected = () => {
```

File: libs/application/templates/children-residence-change/src/fields/Shared.treat.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { style } from 'treat'
 
-export const container = style({
+export const descriptionTextTopContainer = style({
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/Shared.treat.ts:3]: Just thinking if this should be named differently, something like `descriptionOffset` or something that is more descriptive. One might think that this container was needed for all descriptions in all custom field components. I Might be wrong though 🤔

File: libs/application/templates/children-residence-change/src/fields/components/index.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 export { default as ContactInfoRow } from './ContactInfoRow'
-export { default as CopyUrl } from './CopyUrl/CopyUrl'
-export { default as DescriptionText } from './DescriptionText/DescriptionText'
-export { default as InfoBanner } from './InfoBanner/InfoBanner'
-export { default as TransferOverview } from './TransferOverview/TransferOverview'
-export { default as BorderedAccordion } from './BorderedAccordion/BorderedAccordion'
-export { default as ContractOverview } from './ContractOverview/ContractOverview'
+export { default as CopyUrl } from './CopyUrl'
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/components/index.ts:2]: 👏

File: libs/application/templates/children-residence-change/src/forms/ParentBForm.ts [hunk 6]
```diff
@@ -25,13 +25,14 @@ export const ParentBForm: Form = buildForm({
...
         buildCustomField({
-          id: 'parentBIntro',
+          id: 'acceptContract',
           title: m.parentBIntro.general.pageTitle,
           component: 'ParentBIntro',
         }),
       ],
     }),
     buildSection({
+      condition: (answers) => answers.acceptContract === 'accept',
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/forms/ParentBForm.ts:35]: Should we create an enum for this `accept` `reject` ?

File: libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx (hunk 1 above)
Comment (Rikkiabb) [reply libs/application/templates/children-residence-change/src/fields/ContractRejected/index.tsx:14]: Definitely, I was also wondering if we could get this any other way when I was writing this but didn't see another way, i'll add a refactor ticket to our board. The ticket: https://app.asana.com/0/1199733748699731/1200295507610131/f

File: libs/application/templates/children-residence-change/src/fields/Shared.treat.ts (hunk 4 above)
Comment (Rikkiabb) [reply libs/application/templates/children-residence-change/src/fields/Shared.treat.ts:3]: Agree, I'll change it

**Review by VilliGunn — COMMENTED** [review]
Nit picks, nice PR 💯

File: libs/application/templates/children-residence-change/src/forms/ContractRejected.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  buildForm,
+  buildSection,
+  Form,
+  FormModes,
+  buildCustomField,
+} from '@island.is/application/core'
+import Logo from '../../assets/Logo'
+import * as m from '../lib/messages'
+
+export const ParentBContractRejected: Form = buildForm({
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/forms/ContractRejected.ts:11]: Like discussed on a call, we can extract this object to a function and pass in the component ID

**Review by VilliGunn — APPROVED** [review]
🌮

**Pull request merged**